### DISCUSSION TITLE
Matching fontsize on figure with fontsize on presentations

### DISCUSSION THREAD

**Discussion by cgadal** [quarto-dev/quarto-cli#7235] [Q&A]
### Description

I am currently moving my presentations from Latex/Beamer to Quarto/RevealJS. I am having trouble controlling the figure sizes, and matching the font sizes between figures and the text in the presentation.

When I was working with Latex/Beamer:

- I know the chosen font, font sizes in pt, and the output page size in cm in LateX
- I used those as input in matplotlib to get figures with the proper sizes, and matching font size saved in PDF

Using Quarto/Revealjs:

As everything works in pixels with Quarto, I do not know how to match the two. I assume there must be some default PPI or DPI somewhere, that I do not know how to deal with ?

Could you please indicate how to reproduce this workflow using Quarto/revealjs to create matplotlib figures saved in SVG with the same font sizes as used in my Quarto presentation ?

Thanks !

**Comment by mcanouil** [comment]
For generated figures, here are the default: <https://quarto.org/docs/computations/execution-options.html#figure-options>.
See also <https://quarto.org/docs/presentations/revealjs/#figure-size>.

Reveal.js default dimensions and font size:
- <https://quarto.org/docs/presentations/revealjs/advanced.html#presentation-size>
- <https://quarto.org/docs/presentations/revealjs/themes.html#fonts>

**Reply by cgadal** [reply]
Thanks !

There must be something else though, as doing

```
## Test

```{python}
#| echo: true

import matplotlib.pyplot as plt
fig, ax = plt.subplots(1, 1, layout='constrained')

a = plt.plot([1, 2, 5], [1, 3, 8], label='test 1')
b = plt.plot([-1, 2, -5], [1, 3, 8], label='test 2')
c = plt.plot([3, 8, 2], [1, 3, 8], label='test 3')

ax.set_xlabel(r'Test $a = b/\mu$')
ax.set_ylabel(r'Test $a = b/\mu$')
plt.show()
```

and the re-including the generated figure in the as `![](talk_files/figure-revealjs/cell-2-output-1.png)`

gives two different HTML outputs. The HMTL figure inserted after the code execution has an imposed `height` and `width` values. Where does it come from ?

**Reply by mcanouil** [reply]
Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You were initially talking about SVG which has no resolution.
Now you are talking about raster graphics (PNG) which has a resolution.
See `fig-dpi` in the documentation.
Also you might want to use the search plugin of the documentation to find the answers you are looking for, *i.e.*, all the links provided were obtained using the search plugin.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by cgadal** [reply]
You are right, this is getting confusing. Let me set up a proper git repo and I will get back to you. Thanks for your help!
